core/app/c/[communitySlug]/forms/[formSlug]/edit/EditFormTitleButton.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+
+import React, { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import type { FormsId } from "db/public";
+import { Button } from "ui/button";
+import { Dialog, DialogContent, DialogFooter, DialogTitle, DialogTrigger } from "ui/dialog";
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from "ui/form";
+import { CircleCheck } from "ui/icon";
+import { Input } from "ui/input";
+import { toast } from "ui/use-toast";
+
+import { didSucceed, useServerAction } from "~/lib/serverActions";
+import { updateForm } from "./actions";
+
+const editFormTitleSchema = z.object({
+	name: z.string().min(1, "Form name is required"),
+});
+
+const EditFormTitleButton = ({ formId, name }: { formId: FormsId; name: string }) => {
+	const [isOpen, setIsOpen] = useState(false);
+
+	const form = useForm<z.infer<typeof editFormTitleSchema>>({
+		resolver: zodResolver(editFormTitleSchema),
+		defaultValues: {
+			name,
```

Comment:
This default doesn't change if you edit the name and then click edit again. But tbh I don't entirely know why, since the name on the form builder page does!

Reply:
https://github.com/user-attachments/assets/e82c19db-55f0-4d30-aa6f-3f8b6be003cb

like this but its not updating to the new name?